cashweb-registry/src/registry.rs

```diff
@@ -948,4 +1039,314 @@ mod tests {
+
+            signed_message.payload = broadcast_message.encode_to_vec();
+            let payload_hash = Sha256::digest(signed_message.payload.clone().into());
+            signed_message.payload_hash = payload_hash.as_slice().to_vec();
+            signed_message.sig = registry
+                .ecc
+                .sign(&seckey, payload_hash.byte_array().clone())
+                .to_vec();
+
+            let (outpoint, value) = utxos.pop().unwrap();
+            let burn_amount = 10_000;
+            let tx_builder = TxBuilder {
+                version: 1,
+                inputs: vec![TxBuilderInput::new(
+                    TxInput {
+                        prev_out: outpoint,
+                        script: Script::default(),
+                        sequence: SequenceNo::finalized(),
+                        sign_data: Some(SignData::new(vec![
+                            SignField::OutputScript(address.script().clone()),
+                            SignField::Value(value),
+                        ])),
+                    },
+                    Box::new(P2PKHSignatory {
+                        seckey: seckey.clone(),
+                        pubkey,
+                        sig_hash_type: SigHashType::ALL_BIP143,
+                    }),
+                )],
+                outputs: vec![
+                    TxBuilderOutput::Leftover(address.script().clone()),
+                    TxBuilderOutput::Fixed(TxOutput {
+                        value: burn_amount,
+                        script: build_commitment_script(
+                            BROADCAST_MESSAGE_LOKAD_ID,
+                            pubkey.array(),
+                            &payload_hash,
+                        ),
+                    }),
+                ],
+                lock_time: 0,
+            };
+            signed_message.burn_amount = burn_amount;
+            let tx = tx_builder.sign(&registry.ecc, 1000, 546)?;
+            signed_message.burn_txs[0].tx = tx.ser().to_vec();
+            signed_message.burn_txs[0].burn_idx = 1;
+            Ok((signed_message, tx))
+        };
+        let (signed_message, tx) = build_signed_message(proto::BroadcastMessage {
+            timestamp: 1236,
+            topic: "your.mom".to_string(),
+            entries: vec![],
+        })?;
+        println!("Here I am {}", line!());
+
+        // pre-broadcast tx works
+        bitcoind
+            .cmd_text("sendrawtransaction", &[tx.ser().hex().into()])
+            .await?;
+        println!("Here I am {}", line!());
+
+        // Mine block: This would make another "sendrawtransaction" of `tx` fail.
+        bitcoind
+            .cmd_text("generatetoaddress", &[1i32.into(), address.as_str().into()])
+            .await?;
+        println!("Here I am {}", line!());
+
+        let result = registry.put_message(&signed_message).await?;
+        println!("Here I am {}", line!());
+
+        assert_eq!(
```

Comment:
Some of this stuff should probably directly check the verification functions instead of repeating them for both